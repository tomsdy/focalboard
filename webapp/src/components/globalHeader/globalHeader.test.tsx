// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react'
import {Provider as ReduxProvider} from 'react-redux'

import {render} from '@testing-library/react'

import {IntlProvider} from 'react-intl'

import configureStore from 'redux-mock-store'

import GlobalHeader from './globalHeader'

const wrapIntl = (children: any) => <IntlProvider locale='en'>{children}</IntlProvider>

describe('components/sidebar/GlobalHeader', () => {
    const mockStore = configureStore([])
    let store = mockStore({})
    beforeEach(() => {
        store = mockStore({})
    })
    test('header menu should match snapshot', () => {
        const component = wrapIntl(
            <ReduxProvider store={store}>
                <GlobalHeader/>
            </ReduxProvider>,
        )

        const {container} = render(component)
        expect(container).toMatchSnapshot()
    })
})
