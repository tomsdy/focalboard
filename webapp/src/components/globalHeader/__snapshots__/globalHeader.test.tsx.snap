// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/sidebar/GlobalHeader header menu should match snapshot 1`] = `
<div>
  <div
    class="GlobalHeaderComponent"
  >
    <span
      class="spacer"
    />
    <a
      class="GlobalHeaderComponent__button help-button"
      href="https://www.focalboard.com/guide/user?utm_source=webapp"
      rel="noreferrer"
      target="_blank"
    >
      <i
        class="CompassIcon icon-help-circle-outline HelpIcon"
      />
    </a>
    <div
      class="GlobalHeaderSettingsMenu"
    >
      <div
        class="MenuWrapper"
      >
        <div
          class="GlobalHeaderComponent__button menu-entry"
        >
          <i
            class="CompassIcon icon-settings-outline SettingsIcon"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
